bigtable_automation/gcf/bt_trigger.go

```diff
@@ -6,311 +20,237 @@
+		err = adminClient.CreateTable(dctx, tableID)
+		if err != nil {
+			log.Printf("Error creating table %s, retry...", err)
+		} else {
 			ok = true
 			break
 		}
-		time.Sleep(1 * time.Minute)
+		time.Sleep(10 * time.Second)
 	}
 	if !ok {
-		log.Printf("Unable to create table: %s, got error: %v", tableID, err)
-		return err
+		return errors.Errorf("Unable to create table: %s, got error: %v", tableID, err)
 	}
-
 	// Create table columnFamily.
 	log.Printf("Creating column family %s in table %s/%s", columnFamily, btInstance, tableID)
 	if err := adminClient.CreateColumnFamily(dctx, tableID, columnFamily); err != nil {
-		log.Printf("Unable to create column family: csv for table: %s, got error: %v", tableID, err)
-		return err
+		return errors.WithMessagef(err, "Unable to create column family: csv for table: %s, got error: %v", tableID)
 	}
 	return nil
 }
 
-func scaleBT(ctx context.Context, btProjectID, btInstance string, btClusters []string, numNodes int32) error {
+func scaleBT(ctx context.Context, projectID, instance, cluster string, numNodes int32) error {
 	// Scale up bigtable cluster. This helps speed up the dataflow job.
 	// We scale down again once dataflow job completes.
-	instanceAdminClient, err := bigtable.NewInstanceAdminClient(ctx, btProjectID)
+	instanceAdminClient, err := bigtable.NewInstanceAdminClient(ctx, projectID)
 	dctx, cancel := context.WithDeadline(ctx, time.Now().Add(10*time.Minute))
 	defer cancel()
 	if err != nil {
-		log.Printf("Unable to create a table instance admin client. %v", err)
-		return err
+		return errors.Wrap(err, "Unable to create a table instance admin client")
 	}
-	for _, c := range btClusters {
-		log.Printf("Scaling BT %s cluster %s to %d nodes", btInstance, c, numNodes)
-		if err := instanceAdminClient.UpdateCluster(dctx, btInstance, c, numNodes); err != nil {
-			log.Printf("Unable to resize bigtable cluster %s to %d: %v", c, numNodes, err)
-			return err
-		}
+	log.Printf("Scaling BT %s cluster %s to %d nodes", instance, cluster, numNodes)
+	if err := instanceAdminClient.UpdateCluster(dctx, instance, cluster, numNodes); err != nil {
+		return errors.WithMessagef(err, "Unable to resize bigtable cluster %s to %d: %v", cluster, numNodes)
 	}
 	return nil
 }
 
-func parsePath(path string) (string, string, error) {
-	parts := strings.Split(path, "/")
-	if len(parts) != 3 {
-		return "", "", status.Errorf(codes.Internal, "Unexpected number of parts %s", path)
+func btImportControllerInternal(ctx context.Context, e GCSEvent) error {
+	projectID := os.Getenv("projectID")
+	instance := os.Getenv("instance")
+	cluster := os.Getenv("cluster")
+	nodesHigh := os.Getenv("nodesHigh")
+	nodesLow := os.Getenv("nodesLow")
+	dataflowTemplate := os.Getenv("dataflowTemplate")
+	dataPath := os.Getenv("dataPath")
+	controlPath := os.Getenv("controlPath")
+	// Get low and high nodes number
+	nodesH, err := strconv.Atoi(nodesHigh)
+	if err != nil {
+		return err
```

Comment:
Wrap in user-readable error... e.g., say "invalid nodesHigh value" or something?

Reply:
Done